backend/app/controllers/spree/admin/products_controller.rb

```diff
@@ -100,11 +100,15 @@ def collection
 
         params[:q][:s] ||= "name asc"
         @collection = super
-        if params[:q].delete(:deleted_at_null) == '0'
+        # Don't delete params[:q][:deleted_at_null] here because it is used in view to check the
+        # checkbox for 'q[deleted_at_null]'. This also messed with pagination when deleted_at_null is checked.
+        if params[:q][:deleted_at_null] == '0'
           @collection = @collection.with_deleted
         end
         # @search needs to be defined as this is passed to search_form_for
-        @search = @collection.ransack(params[:q])
+        # Temporarily remove params[:q][:deleted_at_null] from params[:q] to ransack products.
+        # This is to include all products and not just deleted products.
+        @search = @collection.ransack(params[:q].reject {|k, v| k == :deleted_at_null})
```

Comment:
Unused block argument - `v`. If it's necessary, use `_` or `_v` as an argument name to indicate that it won't be used.<br>Space between { and | missing.<br>Space missing inside }.